src/figuresClasses.ts

```diff
@@ -1,11 +1,76 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides of triangle should be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('One of three sides is bigger than sum of two others');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
 
-export class Triangle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    private radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius should be greater than 0');
+    }
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    const area = Math.PI * (this.radius * this.radius);
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle' = 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    private width: number,
+    private height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Sides of rectangle should be greater than 0');
+    }
+  }
```

Comment:
you can create function like

function checkOnError(...args: number[]): void {
  if (args.some((arg: number) => arg <= 0)) {
    throw new Error('Some side less 0');
  }
}
It is needed because you have the same logic in three places.